harvardsettings_pi.php

```diff
@@ -382,6 +389,37 @@ public function get_pds_emails($url) {
         return $return_obj;    
     }
     
+    public function get_pds_name_by_email($email) {
+        $url = "/people?email=" . $email . "&filter=name,email";
+        $c = curl_init();
+        curl_setopt($c, CURLOPT_URL, $this->CI->config->item('pds_base_url').$url);
+        curl_setopt($c, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt(
+            $c,
+            CURLOPT_USERPWD,
+            $this->CI->config->item('pds_client_id') . ":" . $this->CI->config->item('pds_client_secret')
+        );
+        $output = json_decode(curl_exec($c));
```

Comment:
Note that [curl_exec()](https://www.php.net/curl_exec) returns `FALSE` if there's a failure of some kind, which will output as `NULL` after being json decoded. Might be worth adding a check for that.